src/components/Translate/TranslateForm.vue

```diff
@@ -0,0 +1,240 @@
+						@update:value="onValueChange('max_tokens', $event)" />
+					<TaskTypeField v-if="showAdvanced"
+						field-key="model"
+						:field="translateTaskType.optionalInputShape.model"
+						:value="inputs?.model ?? null"
+						:options="translateTaskType.optionalInputShapeEnumValues.model"
+						:is-output="false"
+						@update:value="onValueChange('model', $event)" />
+				</div>
+				<div class="col">
+					<TaskTypeField
+						field-key="target_language"
+						:field="translateTaskType.inputShape.target_language"
+						:value="inputs?.target_language ?? null"
+						:options="translateTaskType.inputShapeEnumValues.target_language"
+						:is-output="false"
+						@update:value="onValueChange('target_language', $event)" />
+					<TaskTypeField
+						field-key="output"
+						:field="translateTaskType.outputShape.output"
+						:value="outputs?.output ?? null"
+						:is-output="true"
+						@update:value="onOutputValueChange('output', $event)" />
+					<NcNoteCard v-if="outputEqualsInput"
+						class="warning-note"
+						type="warning">
+						{{ t('assistant', 'The task ran successfully but the result is identical to the input.') }}
+					</NcNoteCard>
+					<NcNoteCard v-else-if="hasInitialOutput"
+						class="warning-note"
+						type="warning">
+						{{ t('assistant', 'This output was generated by AI. Make sure to double-check and adjust.') }}
+					</NcNoteCard>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+
+import TaskTypeField from '../fields/TaskTypeField.vue'
+
+export default {
+	name: 'TranslateForm',
+
+	components: {
+		NcNoteCard,
+		TaskTypeField,
+	},
+
+	props: {
+		translateTaskId: {
+			type: [Number, null],
+			default: null,
+		},
+		translateTaskType: {
+			type: Object,
+			required: true,
+		},
+		inputs: {
+			type: [Object, null],
+			default: null,
+		},
+		outputs: {
+			type: [Object, null],
+			default: null,
+		},
+		showAdvanced: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	emits: [
+		'update:inputs',
+	],
+
+	data() {
+		return {
+		}
+	},
+
+	computed: {
+		outputEqualsInput() {
+			if (typeof this.inputs?.input === 'string' && typeof this.outputs?.output === 'string') {
+				return this.hasInitialOutput && this.outputs.output?.trim() === this.inputs.input?.trim()
+			}
+			return false
+		},
+		hasInitialOutput() {
+			return !!this.outputs?.output?.trim()
+		},
+	},
+
+	watch: {
+		translateTaskType() {
+			console.debug('[assistant] watch translateTaskType', this.translateTaskType, this.translateTaskTypeId)
+			this.setDefaultValues(true)
+		},
+	},
+
+	mounted() {
+		console.debug('[assistant] mounted TranslateForm', this.translateTaskId, this.translateTaskType)
+		// don't set the default values if there is a loaded task (initial or from history)
+		if (this.translateTaskId === null) {
+			this.setDefaultValues(false)
+		}
+
+		this.autosize()
+	},
+
+	methods: {
+		autosize() {
```

Comment:
Not sure this is necessary. If we never call autosize, both text fields behave like in the generic form, they grow vertically (and independently) until they reach their max height. I think I prefer this. Wdyt?

Reply:
My main issue with the current state is that the output field does not grow if the input one contains few lines. But yeah it's kinda fine like it is.